duckduckgo/iOS
# optimise content blocker rules compilation

Task/Issue URL: https://app.asana.com/0/414709148257752/1200055714461291
Tech Design URL:
CC:

**Description**:

Optimisation of content blocker rules compilation logic.  Basically when protection changes, or there's new configuration, recompile the rules, *then* send the notification. 

**Steps to test this PR**:
1. Launch the app, open a few tabs, navigate to some pages, change the protection on one of them.
1. Clean install, launch the app as a background fetch, then launch the app normally.
1. Ensure blocking occurs as expected (JS Console log output).


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
**When ready for review, remember to post the PR in MM**
